test.sh

```diff
@@ -5,10 +5,25 @@ cd java ; fish-pepper ; cd ..
 
 docker build java/images/jboss/ -t fabric8/s2i-java
 
-s2i build --copy java/images/example fabric8/s2i-java fabric8/s2i-java-example
+# Initially intentionally building without incremental
+s2i build --copy java/example fabric8/s2i-java fabric8/s2i-java-example
 
-docker run -p 8080:8080 fabric8/s2i-java-example
+# Now rebuild incrementally, it should not re-download .m2
+s2i build --copy java/example fabric8/s2i-java fabric8/s2i-java-example --incremental
 
-# TODO how to do simplest possible HTTP GET http://localhost:8080 and fail/pass?
-# TODO how to best stop docker again
-# kill $!
+CONTAINER_ID=$(docker run -d -p 8080:8080 fabric8/s2i-java-example)
```

Comment:
Not really sure in which context this test is run, but are you sure that 8080 on localhost is always free ? wouldn't it make sense to use a dynamic portmapping ?

Reply:
Also what is if the docker daemon is running remotely ? (again not sure in which context this script is supposed to be running)